src/Main/Autowiring.php

```diff
@@ -1,335 +1,337 @@
+	 *
+	 * @param array $interfaces List of class interfaces.
+	 *
+	 * @return boolean
+	 */
+	protected function isServiceClass(array $interfaces = [] ): bool
+	{
+		foreach ($interfaces as $interface) {
+			$items = explode('\\', $interface);
+			if (end($items) !== 'ServiceInterface') {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Builds the dependency tree for a single class ($relevantClass)
+	 *
+	 * @param  string $relevantClass       Class we're building dependency tree for.
+	 * @param  array  $filenameIndex       Filename index. Maps filenames to class names.
+	 * @param  array  $classInterfaceIndex Class interface index. Maps classes to interfaces they implement.
+	 * @return array
+	 */
+	protected function buildDependencyTree(string $relevantClass, array $filenameIndex, array $classInterfaceIndex )
+	{
+		$dependencyTree = [];
+		$reflClass      = new \ReflectionClass($relevantClass);
+
+		// If this class has dependencies, we need to figure those out. Otherwise
+		// we just add it to the dependency tree as a class without dependencies.
+		if (! empty($reflClass->getConstructor())) {
+
+			// Go through each constructor parameter.
+			foreach ($reflClass->getConstructor()->getParameters() as $reflParam) {
+
+				if ($reflParam->getType() === null) {
+					continue;
+				}
+
+				$classname         = $reflParam->getType()->getName();
+				$reflClassForParam = new \ReflectionClass($classname);
+
+				// If the expected type is interface, try guessing based on var name. Otherwise
+				// Just inject that class.
+				if ($reflClassForParam->isInterface()) {
+					$matchedClass = $this->tryToFindMatchingClass($reflParam->getName(), $classname, $filenameIndex, $classInterfaceIndex);
+
+					// If we're unable to find exactly 1 class for whatever reason, just skip it, the user
+					// will have to define the dependencies manually.
+					if (empty($matchedClass)) {
+						continue;
+					}
+
+					$dependencyTree[$relevantClass][$matchedClass] = [];
+				} else {
+					$dependencyTree[$relevantClass][$classname] = [];
+				}
+			}
+		} else {
+			$dependencyTree[$relevantClass] = [];
+		}
+
+		return $dependencyTree;
+	}
+
+	/**
+	 * Returns all classes in namespace.
+	 *
+	 * @param  string $namespace    Name of namespace.
+	 * @param  array  $psr4Prefixes Array of psr-4 compliant namespaces and their accompanying folders.
+	 * @return array
+	 */
+	protected function getClassesInNamespace(string $namespace, array $psr4Prefixes ): array
+	{
+		$classes            = [];
+		$namespaceWithSlash = "{$namespace}\\";
+		$pathToNamespace    = $psr4Prefixes[$namespaceWithSlash][0] ?? '';
+
+		if (! is_dir($pathToNamespace)) {
+			return [];
+		}
+
+		$it = new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($pathToNamespace));
+		foreach ($it as $file) {
+			if ($file->isDir()) {
+				continue;
+			}
+			if (preg_match('/\.(php)$/', $file->getFileName()) && preg_match('/^[A-Z]{1,}.*$/m', $file->getFileName())) {
```

Comment:
I think you should be able to do it as a single regex experssion like so (all filename starting with capital letter, ending with `.php`)

```suggestion
			if (preg_match('/[A-Z]{1}.*.php/', $file->getFileName())) {
```

Reply:
Tnx I have implemented the fix